dspace-server-webapp/src/test/java/org/dspace/app/rest/RelationshipRestRepositoryIT.java

```diff
@@ -2673,4 +2684,118 @@ public void orgUnitLeftMaxCardinalityTest() throws Exception {
+            // since the rightMinCardinality constraint would be violated upon deletion
+            isVolumeOfJournal.setRightMinCardinality(0);
+        }
+
+        // Create virtual metadata fields if needed
+        MetadataSchema journalSchema = metadataSchemaService.find(context, "journal");
+        if (journalSchema == null) {
+            journalSchema = metadataSchemaService.create(context, "journal", "journal");
+        }
+        MetadataField journalTitleField = metadataFieldService.findByString(context, "journal.title", '.');
+        if (journalTitleField == null) {
+            journalTitleField = metadataFieldService.create(context, journalSchema, "title", null, "Journal Title");
+        }
+
+        // Create entity items
+        Item journal =
+            ItemBuilder.createItem(context, col1).withRelationshipType("Journal").withTitle("Journal").build();
+        Item journalVolume =
+            ItemBuilder.createItem(context, col1).withRelationshipType("JournalVolume").withTitle("JournalVolume")
+                       .build();
+        Item journalIssue =
+            ItemBuilder.createItem(context, col1).withRelationshipType("JournalIssue").withTitle("JournalIssue")
+                       .build();
+        Item publication =
+            ItemBuilder.createItem(context, col1).withRelationshipType("Publication").withTitle("Publication").build();
+
+        // Link Publication-Journal Issue
+        RelationshipBuilder.createRelationshipBuilder(context, journalIssue, publication, isPublicationOfJournalIssue)
+                           .build();
+        // Link Journal Issue-Journal Volume
+        RelationshipBuilder.createRelationshipBuilder(context, journalVolume, journalIssue, isIssueOfJournalVolume)
+                           .build();
+        mockSolrSearchCore.getSolr().commit(false, false);
+
+        // Verify Publication item via REST does not contain virtual md journal.title
+        getClient().perform(get("/api/core/items/" + publication.getID()))
+                   .andExpect(status().isOk())
+                   .andExpect(jsonPath("$.metadata.journal.title").doesNotExist());
+
+        // Verify Publication item via Solr does not contain virtual md journal.title
+        SolrQuery solrQuery = new SolrQuery();
+        solrQuery.setQuery("search.resourceid:" + publication.getID());
+        QueryResponse queryResponse = mockSolrSearchCore.getSolr().query(solrQuery);
+        assertThat(queryResponse.getResults().size(), equalTo(1));
+        assertTrue(queryResponse.getResults().get(0).getFieldValues("journal.title") == null);
+
+        // Link Journal Volume - Journal
+        RelationshipBuilder.createRelationshipBuilder(context, journal, journalVolume, isVolumeOfJournal).build();
+        mockSolrSearchCore.getSolr().commit(false, false);
+
+        // Verify Publication item via REST does contain virtual md journal.title
+        getClient().perform(get("/api/core/items/" + publication.getID()))
```

Comment:
Shouldn't we also verify that every object in between the publication and the journal *also* contains the virtual metadata `journal.title`?  That might be a more complete test that your recursion is working properly. As, I believe both Journal Volume & Journal Issues should also have `journal.title`...correct?

Reply:
This has been included, and it revealed the JournalVolume metadata was not indexed correctly when creating the relationship. This has been fixed as well